posts/saving-and-analyzing-trending-topics-on-twitter-using-aws-athena-lambda-and-cdk.md

```diff
@@ -0,0 +1,198 @@
+---
+authors:
+- Theo Lebrun
+tags:
+- Twitter
+- AWS
+- AWS Lambda
+- AWS Athena
+- AWS CDK
+date: 2020-08-03T14:50:55.000Z
+title: "Saving and analyzing trending topics on Twitter using AWS Athena, Lambda and CDK"
+image:
+---
+
+With more than 300 million active users, Twitter is still the best platform to provide efficient and accurate real time news. The SMS of the Internet is also one of the best social network and is particularly used a lot by all kind of developers which makes it a very good source of tech information.
+
+With #Hashtags, Twitter found a way to let the users group posts together and follow specific topics in order to get immediate updates. My favorite tool to give you the best Twitter experience is [TweetDeck](https://tweetdeck.twitter.com/), you can define your own custom dashboard in order to fulfill all your needs. My favorite "column" is the `Trending` one and my guilty pleasure is to check it multiple times a day to always be aware of what is happening (even if sometimes I am quite surprised of what is currently trending...).
+
+The goal of this blog is to explain how cheap and easy it is to save and analyze what is trending on Twitter using services provided by AWS. There are a lot of neat use cases that can be done once you have this architecture up and running. One for example would be to notify you if a specific topic is trending in order to not miss important news.
+
+# AWS Services and Architecture
+
+The beauty of AWS is that it provides us all the services that we need in order to build our Twitter analytics platform. That saves us a lot of time and headaches since we can focus on the fun parts of this project instead of less fun parts like hosting and configuration.
+
+The architecture leverages mature AWS services and the flow is pretty simple, here is a diagram to summarize:
+
+![Diagram](https://raw.githubusercontent.com/falydoor/blog-usa/twitter-trends/images/2020/08/twitter-trends-diagram.png)
+
+A future feature that could be added would be to add a notification system that will notify other components if a specific topic is trending. You could for example receive a notification on your phone, a text or simply an email when a defined topic is trending in your location. The lambda will be in charge of checking if one of the trending topic is part of a user defined list and then push a message to AWS SNS for example.
+
+# Deployment using AWS CDK
+
+[AWS Cloud Development Kit](https://aws.amazon.com/cdk/) is THE framework that you need to use to define your AWS cloud infrastructure in code. You can pick your favorite language out of the 5 supported languages (JavaScript, TypeScript, Python, Java or .NET) and its simplicity will save you a lot of time. I recommend using TypeScript since the CDK team prioritizes it and you will also have more examples available on this [GitHub repository](https://github.com/aws-samples/aws-cdk-examples).
```

Comment:
Out of the five supported languages (JavaScript, TypeScript, Python, Java or .NET), you can pick your favorite, and its simplicity will save you time.